webapp/src/ts/services/telemetry.service.ts

```diff
@@ -73,14 +82,26 @@ export class TelemetryService {
     });
   }
 
-  private submitIfNeeded(db) {
-    const monthStart = moment().startOf('month');
-    const dbDate = moment(this.getLastAggregatedDate());
+  // moment when the aggregation starts (the beginning of the current day)
+  private aggregateStartsAt() {
+    return moment().startOf('day');
+    //return moment().startOf('minute');
+  }
 
-    if (dbDate.isBefore(monthStart)) {
-      return this
-        .aggregate(db)
-        .then(() => this.reset(db));
+  // if there is telemetry data to aggregate and is before the current date,
+  // aggregation is performed and the data destroyed
+  private submitIfNeeded(db) {
+    const startOf = this.aggregateStartsAt();
+    const dbDate = moment(this.getFirstAggregatedDate());
+
+    if (dbDate.isBefore(startOf)) {
+      return db.info()
+        .then(info => {
+          if (info.doc_count > 0) {
```

Comment:
> Also, is there ever any case when we don't have any telemetry? We create the database as a result of a record call, which should create at least ONE record, right?

Right, although the aggregation is now executed first, that warranties that at least one record is recorded in the new db that is created after destroying and creating the new db for telemetry. I added that check when i was doing the refactor and I forget to check if it was really needed, because even if there are gaps between records (days that the app is not used), the db will still contain at least one record from the last time aggregation performed. I'll remove the check.